src/drivers/pic/pic.zig

```diff
@@ -89,6 +93,25 @@ pub fn remap(offset1: u8, offset2: u8) void {
     cpu.outb(.pic_slave_data, 0b1111_1111);
 }
 
-pub inline fn ack() void {
-    cpu.outb(0x20, 0x20);
+pub inline fn ack(irq: IRQ) void {
+    if (@intFromEnum(irq) < 8) {
+        cpu.outb(0x20, 0x20);
+    } else {
+        cpu.outb(0xa0, 0x20);
+    }
+}
+
+pub fn get_irq_from_interrupt_id(id: u8) IRQ {
+    if (id >= offset_master and id < offset_master + 8) {
+        return @enumFromInt(id - offset_master);
+    } else if (id >= offset_slave and id < offset_slave + 8) {
+        return @enumFromInt(id - offset_slave);
+    } else {
+        return .Unknown;
```

Comment:
an error would be better

Reply:
Done, I removed the Unknown value from IRQ enum, and display a compile error message instead